app/assets/javascripts/discourse/app/controllers/preferences/interface.js

```diff
@@ -185,6 +185,48 @@ export default class InterfaceController extends Controller {
     return userOptionTextSize !== selectedTextSize;
   }
 
+  _isInLightMode() {
+    return (
+      this.interfaceColor.colorModeIsLight ||
+      (this.interfaceColor.colorModeIsAuto &&
+        !window.matchMedia("(prefers-color-scheme: dark)").matches)
+    );
+  }
+
+  _isInDarkMode() {
+    return (
+      this.interfaceColor.colorModeIsDark ||
+      (this.interfaceColor.colorModeIsAuto &&
+        window.matchMedia("(prefers-color-scheme: dark)").matches)
+    );
+  }
+
+  _shouldEnablePreview(isDarkMode) {
+    return (
+      this.isViewingOwnProfile &&
+      (isDarkMode ? this._isInDarkMode() : this._isInLightMode())
+    );
+  }
+
+  _resolveThemeDefaultColorScheme(colorSchemeId, isDark = false) {
```

Comment:
if this is called once (and with this arg given) there's no need for a default value, I think?

```suggestion
  _resolveThemeDefaultColorScheme(colorSchemeId, isDark) {
```